test/test_utils/ec2.py

```diff
@@ -298,13 +298,24 @@ def launch_instance(
     return response["Instances"][0]
 
 
+def get_available_reservation(ec2_client, instance_type, min_availability=1):
+    reservations = ec2_client.describe_capacity_reservations()
+    for reservation in reservations["CapacityReservations"]:
+        if (
+            reservation["InstanceType"] == instance_type
+            and reservation["AvailableInstanceCount"] >= min_availability
+        ):
+            return reservation
+    return None
+
+
 @retry(
     reraise=True,
     stop=stop_after_delay(30 * 60),  # Keep retrying for 30 minutes
     wait=wait_random_exponential(min=60, max=5 * 60),  # Retry after waiting 1-5 minutes
 )
 def launch_instances_with_retry(
-    ec2_resource, availability_zone_options, ec2_create_instances_definition
+    ec2_resource, ec2_client, availability_zone_options, ec2_create_instances_definition
```

Comment:
[nit] If we're updating the function signature we should update the function docstring with the additional param.

Reply:
good catch, will do